cpp/src/arrow/memory_pool_jemalloc.cc

```diff
@@ -151,6 +151,26 @@ Status jemalloc_set_decay_ms(int ms) {
   return Status::OK();
 }
 
+Status jemalloc_mallctl(const char* name, void* oldp, size_t* oldlenp, void* newp,
+                        size_t newlen) {
+#ifdef ARROW_JEMALLOC
+  int res = mallctl(name, oldp, oldlenp, newp, newlen);
+  return res ? Status::UnknownError(res) : Status::OK();
```

Comment:
use std::strerror like jemalloc_set_decay_ms does above?

Also, maybe rewrite set_decay_ms in terms of this function?

Reply:
Better, use `IOErrorFromErrno`.

Reply:
Done.